examples/basic/ipc_shm_osmemprov_consumer.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <umf/providers/provider_os_memory.h>
+
+#define INET_ADDR "127.0.0.1"
+#define MSG_SIZE 256
+#define RECV_BUFF_SIZE 1024
+
+// consumer's response message
+#define CONSUMER_MSG                                                           \
+    "This is the consumer. I just wrote a new number directly into your "      \
+    "shared memory!"
+
+int main(int argc, char *argv[]) {
+    struct sockaddr_in consumer_addr;
+    struct sockaddr_in producer_addr;
+    char consumer_message[MSG_SIZE];
+    char recv_buffer[RECV_BUFF_SIZE];
+    int producer_socket;
+    int producer_addr_len;
+    int consumer_socket;
+    int ret = -1;
+
+    if (argc < 2) {
+        fprintf(stderr, "usage: %s port\n", argv[0]);
+        return -1;
+    }
+
+    int port = atoi(argv[1]);
+
+    umf_memory_provider_handle_t OS_memory_provider = NULL;
+    umf_os_memory_provider_params_t os_params;
+    enum umf_result_t umf_result;
+
+    os_params = umfOsMemoryProviderParamsDefault();
+    os_params.flag = UMF_MEM_MAP_SHARED;
+
+    // create OS memory provider
+    umf_result = umfMemoryProviderCreate(umfOsMemoryProviderOps(), &os_params,
+                                         &OS_memory_provider);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[consumer] ERROR: creating OS memory provider failed\n");
+        return -1;
+    }
+
+    // get the size of the IPC handle
+    size_t IPC_handle_size;
+    umf_result =
+        umfMemoryProviderGetIPCHandleSize(OS_memory_provider, &IPC_handle_size);
```

Comment:
I think it would be better to do a test based on these files rather than an example. It makes no sense to call Memory Provider IPC API from the user code

Reply:
Done